### PR TITLE
ADOPTED: Datum softdels 

### PR DESCRIPTION
**Pull request by amylizzle**
Adopted from #1776 

Seems to have dropped 2Gb from the memory usage of goon on cog2.

Draft because some mildly weird stuff needs investigating first

### CODE REVIEW

**Comment by amylizzle** [comment]
Running goon with *only* softdels (ie, the whole `ifdef opendream del()` thing disabled) reduces memory usage by the same amount. Using goon's debug tools for softdels I get this for OD:
image
compared to this for BYOND:
image

File: OpenDreamRuntime/DreamManager.Connections.cs [hunk 1]
```diff
@@ -358,3 +358,24 @@ public void Execute(IConsoleShell shell, string argStr, string[] args) {
         dreamManager.HotReloadResource(args[0]);
     }
 }
+
+
+public sealed class ForceGarbageCollectionCommand : IConsoleCommand {
```
Comment (wixoaGit) [inline OpenDreamRuntime/DreamManager.Connections.cs:362]: This is already a server command provided by RT, you can run `gc` in the server's console.

File: OpenDreamRuntime/Procs/DMProc.cs [hunk 2]
```diff
@@ -543,6 +543,7 @@ public override void Dispose() {
             _stack = null;
 
             _dreamValuePool.Return(_localVariables);
+            Array.Clear(_localVariables);
```
Comment (wixoaGit) [inline OpenDreamRuntime/Procs/DMProc.cs:546]: `ArrayPool.Return()` has a `clearArray` parameter.
```suggestion
            _dreamValuePool.Return(_localVariables, true);
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamRuntime/DreamManager.Connections.cs (hunk 1 above)
Comment (amylizzle) [reply OpenDreamRuntime/DreamManager.Connections.cs:362]: oh no worries, I'll delete

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**